### PR TITLE
Feat pdf parsing

### PR DESCRIPTION
**Pull request by brianzhouzc**
Added resume processing

### CODE REVIEW

**Comment by brianzhouzc** [comment]
image
Everything should be working. Users can edit, delete, or add skills on the profile page.
I made a new profile page in React and got rid of the old django page.

**Comment by liewshanhong** [comment]
On Profile page, should we redirect users to questions if they are left unanswered? Also, should we just display the top card(email, age,...) when the users haven't uploaded their resume?

**Comment by brianzhouzc** [comment]
> On Profile page, should we redirect users to questions if they are left unanswered? Also, should we just display the top card(email, age,...) when the users haven't uploaded their resume?

Good idea, I'll redirect the user to the questions page if they haven't answered the questions. 

I think we should still leave the skills card on the profile page even if they haven't uploaded any resume. They can still use it to manually add skills if they wish.

**Comment by brianzhouzc** [comment]
> > On Profile page, should we redirect users to questions if they are left unanswered? Also, should we just display the top card(email, age,...) when the users haven't uploaded their resume?
> 
> Good idea, I'll redirect the user to the questions page if they haven't answered the questions.
> 
> I think we should still leave the skills card on the profile page even if they haven't uploaded any resume. They can still use it to manually add skills if they wish.

I would do this in the feat-refactor branch. If you don't see any other issue, please approve the PR. You can start a review by clicking the `Files changed` tab on the top

File: skills_DS/react/src/components/Upload.js [hunk 1]
```diff
@@ -20,16 +20,36 @@ const Upload = () => {
...
+  const checkResumeProcessing = () => {
+    try {
+      axios.get("/api/get-profile", {headers: {"X-CSRFTOKEN": Cookies.get("csrftoken")}}).then((response) => {
+        if (!response.data.success.resume_processing) {
+          window.location.href = "profile/";
+        } else {
+          setTimeout(checkResumeProcessing, 2000);
+        }
+      });
+    } catch (err) { }
```
Comment (SanaBob) [inline skills_DS/react/src/components/Upload.js:32]: maybe console.error here

File: skills_DS/react/src/components/Upload.js [hunk 2]
```diff
@@ -20,16 +20,36 @@ const Upload = () => {
...
+          "Content-Type": "multipart/form-data"
+        }
+      }).then((response) => {
+        setButtonClicked(true);
+        checkResumeProcessing();
+      }).catch(err => console.error(err));
 
-      setButtonClicked(true)
-    } catch (err) {}
+    } catch (err) { }
```
Comment (SanaBob) [inline skills_DS/react/src/components/Upload.js:52]: try catch outside seems unnecessary now with .catch

File: skills_DS/react/src/components/Profile.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    const handleAddRow = () => {
+        const newSkills = [...skills];
+        const max_index = Math.max.apply(Math, newSkills.map(function (o) { return o.id; }))
+        newSkills.push({ id: max_index + 1, value: "" });
+        setSkills(newSkills);
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        axios.post("/api/update-user-skills", skills, { headers: { "X-CSRFTOKEN": Cookies.get("csrftoken") } }).then(setSubmitted(true));
```
Comment (SanaBob) [inline skills_DS/react/src/components/Profile.js:39]: .catch on api call

**Review by SanaBob — APPROVED** [review]
some minor suggestions, otherwise looks good

**Review by liewshanhong — APPROVED** [review]
Everything is working.

File: skills_DS/react/src/components/Upload.js (hunk 1 above)
Comment (brianzhouzc) [reply skills_DS/react/src/components/Upload.js:32]: Will be resolve in `feat-refactor`

File: skills_DS/react/src/components/Upload.js (hunk 2 above)
Comment (brianzhouzc) [reply skills_DS/react/src/components/Upload.js:52]: Will be resolve in `feat-refactor`

File: skills_DS/react/src/components/Profile.js (hunk 3 above)
Comment (brianzhouzc) [reply skills_DS/react/src/components/Profile.js:39]: Will be resolve in `feat-refactor`

**Pull request merged**